### PR TITLE
Fix compilation errors when compiling with -Wall -Werror

### PR DESCRIPTION
**Pull request by dbarbier**
`SpecFunc::IncompleteBetaInverse`, `DistFunc::[pq]DickeyFuller*` and `RandomMixture::computeQuantile` methods have an unused `tail` argument which should be fixed too.

### CODE REVIEW

**Comment by jschueller** [comment]
I've also had the new gcc 8 warning Wclass-memaccess (enabled by Wall), any how idea how to fix this one ?
```
In file included from /tmp/openturns/lib/src/Base/Stat/RandomGenerator.cxx:22,
                 from /tmp/openturns/lib/src/cotire/OT_orig_CXX_unity_270_359.cxx:47:
/tmp/openturns/lib/src/Base/Stat/dsfmt.h: In instantiation of ‘void tutils::detail::dsfmt<MEXP>::set_state(OT::UnsignedInteger*) [with int MEXP = 19937; OT::UnsignedInteger = long unsigned int]’:
/tmp/openturns/lib/src/Base/Stat/RandomGenerator.cxx:71:37:   required from here
/tmp/openturns/lib/src/Base/Stat/dsfmt.h:113:21: warning: ‘void* memcpy(void*, const void*, size_t)’ copying an object of non-trivial type ‘class tutils::uint64v2_t’ from an array of ‘uint32_t’ {aka ‘unsigned int’} [-Wclass-memaccess]
               memcpy(&u_[i], &words[0], 16);
               ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In file included from /tmp/openturns/lib/src/Base/Stat/dsfmt.h:29,
                 from /tmp/openturns/lib/src/Base/Stat/RandomGenerator.cxx:22,
                 from /tmp/openturns/lib/src/cotire/OT_orig_CXX_unity_270_359.cxx:47:
/tmp/openturns/lib/src/Base/Stat/simd.h:40:9: note: ‘class tutils::uint64v2_t’ declared here
   class uint64v2_t
```

**Comment by dbarbier** [comment]
@jschueller Ok, this is fixed now.

I had to come back at these changes, everything should be fine now.

File: python/src/ResourceMap.i [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace OT {
 
 %extend ResourceMap {
 
-UnsignedInteger __len__() const
+static UnsignedInteger __len__() const
```
Comment (jschueller) [inline python/src/ResourceMap.i]: ResourceMap.i:22: Error: Static function __len__() const cannot have a qualifier.

File: python/src/IdentityMatrix.i [hunk 2]
```diff
@@ -13,7 +13,7 @@ namespace OT {
 %extend IdentityMatrix {
   IdentityMatrix(const IdentityMatrix & other) { return new OT::IdentityMatrix(other); }
 
-PyObject * __setitem__(PyObject * args, Scalar val)
+static PyObject * __setitem__(PyObject * , Scalar )
```
Comment (jschueller) [inline python/src/IdentityMatrix.i]: Why static ? Is there a warning after swig translates this ?

Comment (dbarbier) [reply python/src/IdentityMatrix.i]: Yes

File: python/src/ResourceMap.i (hunk 1 above)
Comment (jschueller) [reply python/src/ResourceMap.i]: Maybe it only fails with swig master here?

Comment (dbarbier) [reply python/src/ResourceMap.i]: Indeed it does not make sense to have a const qualifier for static methods, I'll drop it.

**Comment by regislebrun** [comment]
+1

**Pull request merged**